### ISSUE TITLE
Fix alignment of opener elements

### ISSUE THREAD

**Issue by joewiz** [HistoryAtState/hsg-shell#261]
A common feature of FRUS document openers is the horizontal alignment of document identifiers and datelines. For example, notice how in [this page](https://history.state.gov/historicaldocuments/frus1936v01/pg_1) from [this document](https://history.state.gov/historicaldocuments/frus1936v01/d1), "No. 1815 Political" is left-aligned on the same line as the dateline, "Geneva, August 17, 1936...":

> <img width="394" alt="0081_png__497x800_" src="https://cloud.githubusercontent.com/assets/59118/20773015/e8b19e46-b71d-11e6-9697-eb91249a5e8c.png">

Since moving to the new hsg site, the alignment of these elements has been broken:

> <img width="631" alt="foreign_relations_of_the_united_states_diplomatic_papers__1936__general__british_commonwealth__volume_i_-_office_of_the_historian" src="https://cloud.githubusercontent.com/assets/59118/20772978/cca400ea-b71d-11e6-95b9-d64a93897b2e.png">

We capture this in TEI as:

```xml
<opener>
    <dateline rendition="#right">
        <placeName>
            <hi rend="smallcaps">Geneva</hi>
        </placeName>, <date when="1936-08-17">August 17,
        1936</date>.<lb/>[Received August 26.]</dateline>
    <seg rendition="#left">No. 1815 Political</seg>
</opener>
<p>
    <hi rend="smallcaps">Sir</hi>:
    <!--snip-->
</p>
```

Where these rendition values are defined as:

```xml
<rendition xml:id="left">display: inline; float: left;</rendition>
<rendition xml:id="right">display: inline; float: right;</rendition>
```

Currently the HTML is being generated as follows:

```html
<div class="tei-opener">
    <div class="tei-dateline">
        <span class="tei-placeName"><span class="tei-hi3">Geneva</span></span>, <span class="tei-date">August 17, 1936</span>.<br/>
        [Received August 26.]
    </div>
    <span class="tei-seg document_left">No. 1815 Political</span>
</div>
<p class="tei-p3">
    <span class="tei-hi3">Sir</span>:
    <!--snip-->
</p>
```

I have started a PR with an attempted solution to this, but it is still not working. My idea was to change the ODD so `<dateline>` elements.  Right now, our ODD defines `<dateline>` as `@behavior="block"` - but I see the case of openers with a child seg-dateline pair as follows a block `<opener>` with inline child `<seg>` and `<dateline>`. The problem, though, is that the `<opener>` seems to have zero height - so the "Sir" line is mangled into the opener line:

> <img width="640" alt="screen shot 2016-11-30 at 5 09 01 pm" src="https://cloud.githubusercontent.com/assets/59118/20773433/bb7ca0b8-b71f-11e6-932a-fcb8a02c38f5.png">

@plutonik-a Could you assist with this, working from the branch I'm about to create?

**Comment by joewiz** [comment]
For additional context, see https://github.com/HistoryAtState/frus/issues/28 where this alignment requirement was first described and added to FRUS TEI.

**Comment by tuurma** [comment]
one option is to reposition `<seg>` before `<dateline>` in the TEI source

if that's unacceptable, something along this lines in frus.odd helps a bit but would require further css adjusments, I think

```                <elementSpec mode="change" ident="dateline">
                    <model predicate="parent::opener" behaviour="inline" useSourceRendition="true">
                    </model>
                    <model behaviour="block">
                        <outputRendition>text-align: right;</outputRendition>
                    </model>
                </elementSpec>
```

**Comment by tuurma** [comment]
mistakenly closed, meant closing #260

**Comment by joewiz** [comment]
@tuurma Thanks for looking into this! Could you post a screenshot of the document in the original report showing the fix? The link to the document in question on your local machine would be <http://localhost:8080/exist/apps/hsg-shell/historicaldocuments/frus1936v01/d1>.

**Comment by joewiz** [comment]
@tuurma Ah, I see! I'll make the similar request over on #260.

**Comment by joewiz** [comment]
@tuurma Scratch that! #260 can't really be captured with a screenshot ;)

**Comment by tuurma** [comment]
current state of tei source:
 
```xml
                <opener>
                    <dateline rendition="#right">
                        <placeName><hi rend="smallcaps">Geneva</hi></placeName>, 
                        <date when="1936-08-17">August 17, 1936</date>.
                        <lb/>[Received August 26.]
                    </dateline>
                    <seg rendition="#left">No. 1815 Political</seg>
                </opener>
```

switching the order of `seg` and `dateline` fixes the issue
```xml
                <opener>
                    <seg rendition="#left">No. 1815 Political</seg>
                    <dateline rendition="#right">
                        <placeName><hi rend="smallcaps">Geneva</hi></placeName>, 
                        <date when="1936-08-17">August 17, 1936</date>.
                        <lb/>[Received August 26.]
                    </dateline>
                </opener>
```


![image](https://cloud.githubusercontent.com/assets/449468/24208753/556881d4-0f1c-11e7-8678-79f3c8cd2973.png)

**Comment by tuurma** [comment]
I think that having this change in the source follows the guidelines proposed in https://github.com/HistoryAtState/frus/issues/28#issuecomment-240739832 which reflect the left-to-right order of text in the printed document anyway but if it's too much hassle we can adjust frus.odd and ask @plutonik-a for some css magic to move things around accordingly. Happy to hear what you think @joewiz

**Comment by joewiz** [comment]
In https://github.com/HistoryAtState/hsg-shell/commit/6b1647411667512f9e8ed3ae884d3a6fbdf5c4a5 I made some changes to the ODD in preparation for a global update to the ordering of right-aligned datelines and left-aligned segs in openers:

- Before, dateline was assumed to be a block element, and was always right-aligned
- Recently, though, we have begun handling dateline’s right alignment via rendition attribute (evaluating to float:right), in order to allow alignment with seg elements (whose rendition attribute correspondingly evaluates to float:left)
- With this commit, we now first check datelines to see if there is a rendition present; if so, we let it through and treat the result as an inline element
- If no rendition attribute is present, we assume the old default behavior is desired, so we make it into a right-aligned block
- Also, this adds a little bottom margin and padding to openers, to ensure these aligned opener elements have space beneath them; since they are inline floats, they tend to let the following block elements up onto their line. This commit avoids that phenomenon pushes the following block elements back down.

**Comment by joewiz** [comment]
In https://github.com/HistoryAtState/hsg-shell/commit/5946ff592a84d9376ff39431133605772e20dee2 I reverted the changes discussed in my previous comment. Forcing datelines to be right-aligned block-level elements overcomes the problem of the next element bleeding into the opener. 

In other commits I completed the swapping of the position of segs and datelines throughout the FRUS corpus.

With this and the other updates to FRUS (now all being pushed to HSG as I type), here is the resulting appearance of [the original document](https://history.state.gov/historicaldocuments/frus1936v01/d1) from this report, showing that the document-identifier seg and the dateline are now properly aligned:

> ![screen shot 2017-03-31 at 12 07 01 pm](https://cloud.githubusercontent.com/assets/59118/24558916/d8f93760-160a-11e7-837a-468fc9535357.png)
